vault/identity_store_aliases.go

```diff
@@ -256,8 +258,46 @@ func (i *IdentityStore) handleAliasUpdateCommon(req *logical.Request, d *framewo
 		return nil, err
 	}
 
+	var existingEntity *identity.Entity
```

Comment:
Don't you need to grab the alias lock before the MemDBAliasByFactors call above? Otherwise what happens if an automatic call is registering an alias at the same time as this manual call?